src/llama_recipes/datasets/alpaca_dataset.py

```diff
@@ -41,30 +43,73 @@ def __len__(self):
     def __getitem__(self, index):
         IGNORE_INDEX = -100  # The default setting in CrossEntropyLoss
 
-
         ann = self.ann[index]
         if ann.get("input", "") == "":
-            prompt = PROMPT_DICT["prompt_no_input"].format_map(ann)
+            user_prompt = PROMPT_DICT["prompt_no_input"].format_map(ann)
+        else:
+            user_prompt = PROMPT_DICT["prompt_input"].format_map(ann)
+
+        # If vocab size is above 128000, use the chat template to generate the tokens as it is from Llama 3 family models
+        if self.tokenizer.vocab_size >= 128000:
+            dialog = [
+                {"role": "user", "content": user_prompt},
+                {"role": "assistant", "content": ann["output"]},
+            ]
+            dialog_tokens = self.tokenizer.apply_chat_template(dialog)
+            eot_indices = [i for i, n in enumerate(dialog_tokens) if n == 128009]
+            labels = copy.copy(dialog_tokens)
+            last_idx = 0
+            # system prompt header "<|start_header_id|>system<|end_header_id|>" has been tokenized to [128006, 9125, 128007]
+            # user prompt header "<|start_header_id|>user<|end_header_id|>" has been tokenized to [128006, 882, 128007]
+            prompt_header_seqs = [[128006, 9125, 128007], [128006, 882, 128007]]
+            for n, idx in enumerate(eot_indices):
+                current_seq = labels[last_idx : idx + 1]
+                if self.check_header(prompt_header_seqs, current_seq):
+                    # found prompt header, indicating that this seq should be masked
+                    labels[last_idx : idx + 1] = [IGNORE_INDEX] * (idx - last_idx + 1)
+                else:
+                    last_idx = idx + 1
```

Comment:
Don't we need to increment the last index in any case?
```suggestion
                last_idx = idx + 1
```